enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/transaction/log/checkpoint/ConfigurableIOLimiter.java

```diff
@@ -32,41 +32,82 @@
 
 public class ConfigurableIOLimiter implements IOLimiter
 {
-    private static final AtomicIntegerFieldUpdater<ConfigurableIOLimiter> disableCountUpdater =
-            AtomicIntegerFieldUpdater.newUpdater( ConfigurableIOLimiter.class, "disabledCount" );
+    private static final AtomicLongFieldUpdater<ConfigurableIOLimiter> stateUpdater =
```

Comment:
Do we still need this? `state` could as well be an `AtomicLong` since we always access if through the updater, right?
Or line 133 should just get the variable without the updater
```
133:     long state = this.state;
```
since I'm guessing the original reason to use the updater is to get rid of the overhead from the `get()` call.

Reply:
It's to avoid the object indirection of the AtomicLong.

Reply:
That's what I mean, you can omit all the `stateUpdater.get()` calls and just read it with an assignment. Then you could get rid of the `SuppressWarning` as well.

Reply:
There's not really an indirection with `stateUpdater` because it's `static final`. I like the symmetry in the code as is.